### PR TITLE
Update sodiumoxide dependency

### PR DESCRIPTION
**Pull request by davidMcneil**
Signed-off-by: David McNeil <mcneil.david2@gmail.com>

### CODE REVIEW

File: components/core/src/crypto.rs [hunk 1]
```diff
@@ -249,6 +251,8 @@ pub mod dpapi;
 pub mod hash;
 pub mod keys;
 
+pub fn init() -> Result<()> { sodiumoxide::init().map_err(|_| Error::SodiumInitFailed) }
```
[the remark this answers is not in the document]
Comment (davidMcneil) [reply components/core/src/crypto.rs:254]: Unfortunately, the `Err` type returned is just a [`()`](https://docs.rs/sodiumoxide/0.2.5/sodiumoxide/fn.init.html).

File: .expeditor/test/fixtures/expected_hab_0.88.0_windows_archive_contents.txt [hunk 2]
```diff
@@ -44,7 +44,6 @@ hab-0.88.0-20191009205851-x86_64-windows/concrt140.dll
 hab-0.88.0-20191009205851-x86_64-windows/hab.exe
 hab-0.88.0-20191009205851-x86_64-windows/libeay32.dll
 hab-0.88.0-20191009205851-x86_64-windows/liblzma.dll
-hab-0.88.0-20191009205851-x86_64-windows/libsodium.dll
```
[the remark this answers is not in the document]
Comment (davidMcneil) [reply .expeditor/test/fixtures/expected_hab_0.88.0_windows_archive_contents.txt]: Yeah, I got a little overzealous deleting stuff :sweat_smile: . Should be added back in now.

File: components/core/src/crypto.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (christophermaier) [reply components/core/src/crypto.rs:254]: Well, _that's_ useful.

**Comment by scotthain** [comment]
This is the best PR I have seen all year. 🚀 🍰

**Pull request merged**